scripts/generateReadmes.js

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+import path from 'path';
+import t from 'tcomb';
+import parse from './parse';
+import template from './template';
+import debug from 'debug';
+
+debug.enable('brc:*');
+const log = debug('brc:generate-readmes');
+
+const checkDocs = (json, fileName) => {
+  log('Checking extracted docs have necessary keys...');
+  if (!json.name) {
+    throw new Error(`Missing name for file ${fileName}`);
+  }
+  if (!json.description) {
+    throw new Error(`Missing description for file ${fileName}.`);
+  }
+
+};
+
+const generateReadMe = (dir, fileName) => {
+  const filePath = path.join(dir, fileName);
+  log(`Extracting docs from ${filePath}`);
+  const json = parse(filePath);
+  log(`Extracted: ${t.stringify(json)}`);
+  checkDocs(json, fileName);
+  const readMeContent = template(json);
+  const readMeFilePath = path.join(dir, 'README.md');
+  log(`Writing README.md for file '${fileName}'`);
+  try {
+    fs.writeFileSync(readMeFilePath, readMeContent, 'utf-8');
+    log(`README.md for ${fileName} created at ${readMeFilePath}!`);
+  } catch (e) {
+    log(`Error creating ${fileName} README.md: ${t.stringify(e.getMessage())}`);
+  }
+};
+
+const walkThroughDirs = (dir, cb) => {
+  if (fs.statSync(dir).isDirectory()) {
+    log(`Walking through ${dir}`);
+    fs.readdir(dir, (err, results = []) => {
+      log(`${dir} contains: ${results.join('\n')}`);
+      results
+      .filter(r => r !== 'index.js' && ['', '.js'].indexOf(path.extname(r)) > -1)
+      .forEach(r => {
+        const pathToCheck = path.join(dir, r);
+        log(`Checking path: ${pathToCheck}`);
+        if (fs.statSync(pathToCheck).isFile() && path.extname(r) === '.js') {
```

Comment:
I have some concerns about this logic.
For which files do we want to create a README.md? I thought only the main component file.
How do you handle folders with multiple components in it? last component wins?
also, if we have a folder like:
```
- /component
  - MainComponent.js
  - /sub-components
    - Header.js
    - Body.js
    - Footer.js
```
are you generating a README.md also for the folder `sub-components`? I think we shouldn't.

If you want developers to document also minor components we can think of simple CI validation check which controls if every component is documented properly (you already implemented the logic for this) without creating README.md files.

what do you think @ascariandrea?